napari/components/viewer_model.py

```diff
@@ -869,8 +869,9 @@ def add_surface(
         data : 3-tuple of array
             The first element of the tuple is an (N, D) array of vertices of
             mesh triangles. The second is an (M, 3) array of int of indices
-            of the mesh triangles. The third element is the (N, ) array of
-            values used to color vertices.
+            of the mesh triangles. The third element is the (N, K0, ..., KL)
+            array of values used to color vertices where the additional L
+            dimensions are used to color the same mesh with different values.
```

Comment:
Oh, I think this shape should be (K0, ..., KL, N)... This would mirror NumPy broadcasting (implicit dimensions are prepended, not appended).

Reply:
changed